Prevent E2E workflow to fail on build phase

# Description

Currently the E2E workflow is breaking, on build phasem more often and in some cases always, preventing to run the tests and merge the PR.

## Context 

Job examples:
- https://github.com/rancher/dashboard/pull/8654 ([attempt 4](https://github.com/rancher/dashboard/actions/runs/4807444086/jobs/8558028969))
-  

Error trace:
```
-  Building for production...
Instrumenting code for coverage

<--- Last few GCs --->

[3322:0x46cb500]   324946 ms: Mark-sweep 2025.3 (2088.6) -> 2018.0 (2081.0) MB, 2898.8 / 0.0 ms  (average mu = 0.111, current mu = 0.030) allocation failure scavenge might not succeed
[3322:0x46cb500]   326935 ms: Mark-sweep 2027.9 (2090.9) -> 2021.3 (2084.3) MB, 1948.4 / 0.0 ms  (average mu = 0.074, current mu = 0.020) allocation failure scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0xa3ad50 node::Abort() [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
 2: 0x970199 node::FatalError(char const*, char const*) [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
 3: 0xbba90e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
 4: 0xbbac87 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
 5: 0xd76ea5  [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
 6: 0xd77a2f  [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
 7: 0xd8586b v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
 8: 0xd8942c v8::internal::Heap::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
 9: 0xd4ebfd v8::internal::Factory::AllocateRaw(int, v8::internal::AllocationType, v8::internal::AllocationAlignment) [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
10: 0xd4ae59 v8::internal::FactoryBase<v8::internal::Factory>::AllocateRawArray(int, v8::internal::AllocationType) [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
11: 0xd4af14 v8::internal::FactoryBase<v8::internal::Factory>::NewFixedArrayWithFiller(v8::internal::Handle<v8::internal::Map>, int, v8::internal::Handle<v8::internal::Oddball>, v8::internal::AllocationType) [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
12: 0xed93be  [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
13: 0xed950d  [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
14: 0x1077ac3 v8::internal::Runtime_GrowArrayElements(int, unsigned long*, v8::internal::Isolate*) [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
15: 0x1449379  [/opt/hostedtoolcache/node/14.21.3/x64/bin/node]
Aborted (core dumped)
error Command failed with exit code [134](https://github.com/rancher/dashboard/actions/runs/4807444086/jobs/8558028969#step:5:135).
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 134.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 134.
```

--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "docker-dev": "docker run --rm --name dashboard-dev -p 8005:8005 -e API=$API -v $(pwd):/src -v dashboard_node:/src/node_modules rancher/dashboard:dev",
     "docker:local:start": "docker run -d --restart=unless-stopped -p 80:80 -p 443:443 -e CATTLE_BOOTSTRAP_PASSWORD=password -e CATTLE_PASSWORD_MIN_LENGTH=3 --name cypress --privileged rancher/rancher:v2.7-head",
     "docker:local:stop": "docker kill cypress || docker rm cypress || true",
-    "build": "yarn lint && ./node_modules/.bin/vue-cli-service build",
+    "build": "yarn lint && NODE_OPTIONS=--max_old_space_size=4096 ./node_modules/.bin/vue-cli-service build",
     "build:lib": "cd pkg/rancher-components && yarn build:lib",
     "analyze": "./node_modules/.bin/vue-cli-service build --report",
     "start": "./node_modules/.bin/vue-cli-service serve",

